repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/datamesh/DomainPreProcessor.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.atlas.type.AtlasTypeRegistry;
+import org.apache.atlas.utils.AtlasPerfMetrics;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+import static org.apache.atlas.repository.Constants.*;
+import static org.apache.atlas.repository.graph.GraphHelper.getActiveChildrenVertices;
+import static org.apache.atlas.repository.store.graph.v2.preprocessor.PreProcessorUtils.*;
+import static org.apache.atlas.repository.util.AccessControlUtils.ATTR_POLICY_RESOURCES;
+import static org.apache.atlas.repository.util.AtlasEntityUtils.mapOf;
+
+public class DomainPreProcessor extends AbstractDomainPreProcessor {
+    private static final Logger LOG = LoggerFactory.getLogger(DomainPreProcessor.class);
+    private AtlasEntityHeader parentDomain;
+    private EntityGraphMapper entityGraphMapper;
+    private EntityMutationContext context;
+
+    public DomainPreProcessor(AtlasTypeRegistry typeRegistry, EntityGraphRetriever entityRetriever,
+                              AtlasGraph graph, EntityGraphMapper entityGraphMapper) {
+        super(typeRegistry, entityRetriever, graph);
+        this.entityGraphMapper = entityGraphMapper;
+    }
+
+    @Override
+    public void processAttributes(AtlasStruct entityStruct, EntityMutationContext context,
+                                  EntityMutations.EntityOperation operation) throws AtlasBaseException {
+        //Handle name & qualifiedName
+        if (operation == EntityMutations.EntityOperation.CREATE && LOG.isDebugEnabled()) {
+            LOG.debug("DomainPreProcessor.processAttributes: pre processing {}, {}",
+                    entityStruct.getAttribute(QUALIFIED_NAME), operation);
+        }
+
+        this.context = context;
+
+        AtlasEntity entity = (AtlasEntity) entityStruct;
+        AtlasVertex vertex = context.getVertex(entity.getGuid());
+
+        setParent(entity, context);
+
+        if (operation == EntityMutations.EntityOperation.CREATE) {
+            processCreateDomain(entity, vertex);
+        }
+    }
+
+    private void processCreateDomain(AtlasEntity entity, AtlasVertex vertex) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("processCreateDomain");
+        String domainName = (String) entity.getAttribute(NAME);
+        String parentDomainQualifiedName = (String) entity.getAttribute(PARENT_DOMAIN_QN);
+
+        domainExists(domainName, parentDomainQualifiedName);
+        entity.setAttribute(QUALIFIED_NAME, createQualifiedName(parentDomainQualifiedName));
+
+        RequestContext.get().endMetricRecord(metricRecorder);
+    }
+
+    public static String createQualifiedName(String parentDomainQualifiedName) {
+        if (StringUtils.isNotEmpty(parentDomainQualifiedName) && parentDomainQualifiedName !=null) {
```

Comment:
no need to check parentDomainQualifiedName !=null, StringUtils.isNotEmpty takes care of it